packages/cli-repl/src/cli-repl.ts

```diff
@@ -500,12 +501,16 @@ export class CliRepl implements MongoshIOProvider {
       } as any /* axiosConfig and axiosRetryConfig are existing options, but don't have type definitions */
     );
     this.toggleableAnalytics = new ToggleableAnalytics(
-      new ThrottledAnalytics({
-        target: this.segmentAnalytics,
-        throttle: {
-          rate: 30,
-          metadataPath: this.shellHomeDirectory.paths.shellLocalDataPath,
-        },
+      new SampledAnalytics({
+        target: new ThrottledAnalytics({
+          target: this.segmentAnalytics,
+          throttle: {
+            rate: 30,
+            metadataPath: this.shellHomeDirectory.paths.shellLocalDataPath,
+          },
+        }),
+        sampling: () =>
+          !!process.env.MONGOSH_ANALYTICS_SAMPLE || Math.random() < 0.3,
```

Comment:
Okay, just to check before approving this, is 30% really the value we got from the analytics team? That sounds much higher than what I was expecting (1% or so)

Reply:
It's not decided yet, it's just a placeholder until the analytics team gives me an answer :D. I was waiting for Gaurab to provide me the correct sampling.

If you want we can keep this PR and close it with the proper percentage or merge it and do a small PR changing the number once it's decided.

No strong opinion here.

Reply:
Let's wait for tomorrow and see if we get a quick answer, and if they mention that they need to think about it more thoroughly I'll merge.